content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+        CreateEntityWithShapeComponent(typeId);
+    });
+    ```
+
+
+
+### Communicate with EBusses
+
+Now, you will define `CreateEntityWithShapeComponent(...)`, which uses communicates with O3DE EBusses to create a new entity with a component specified by the `typeId` parameter.
+
+1. Use the namespace `AzToolsFramework`. This namespace provides APIs to the necessary Editor EBusses. 
+
+2. Send a request to create a new entity by calling `EditorRequestBus::BroadcastResult(...)`. Dispatch the `EditorRequests::CreateNewEntity` event, which instantiates an `AZ::EntityId` and stores it in `newEntityId`. 
+
+3. If the user entered a name in the input field, update the entity's name. 
+
+   - Get the name of the entity by calling `m_nameInput->text()` and store it in `entityName`.
+
+   - If the user enabled the check box to apply a suffix of the component's name to the entity's name, query the component's name. Call `EditorComponentAPIBus::BroadcastResult(...)` and dispatch the `EditorComponentAPIRequests::FindComponentTypeNames` event, which finds the component names of the provided list and stores them in `componentNames`.  
+
+   - Format the name and suffix, if any.
+
+    - Call `EditorEntityAPIBus::Event(...)` and dispatch `EditorEntityAPIRequests::SetName` to set the name of `newEntityId` to `entityName`. 
+
+4. Add a Shape component to the entity. Call `EditorComponentAPIBus::Broadcast(...)` and dispatch `EditorComponentAPIRequests::AddComponentsOfType`, which adds the components from the provided list to `newEntityId`.
+
+```cpp
+    void ShapeExampleWidget::CreateEntityWithShapeComponent(const AZ::TypeId& typeId)
+    {
+        // 1
+        using namespace AzToolsFramework;
+
+        // 2
+        AZ::EntityId newEntityId;
+        EditorRequestBus::BroadcastResult(newEntityId, &EditorRequests::CreateNewEntity, AZ::EntityId());
+
+        // 3
+        if (!m_nameInput->text().isEmpty())
+        {
+            QString entityName = m_nameInput->text();
+
+            if (m_addShapeNameSuffix->isChecked())
+            {
+                AZStd::vector<AZStd::string> componentNames;
+                EditorComponentAPIBus::BroadcastResult(
+                    componentNames,
+                    &EditorComponentAPIRequests::FindComponentTypeNames,
+                    AZ::ComponentTypeList{ typeId }
+                );
+
+                AZ_Assert(componentNames.size() == 1, "Unable to find component name");
+
+                QString shapeName(componentNames[0].c_str());
+                shapeName.replace(" ", "");
+
+                entityName = QString("%1_%2").arg(entityName).arg(shapeName);
+            }
+            
+            EditorEntityAPIBus::Event(newEntityId, &EditorEntityAPIRequests::SetName, entityName.toUtf8().constData());
+        }
+
+        // 4
+        EditorComponentAPIBus::Broadcast(&EditorComponentAPIRequests::AddComponentsOfType, newEntityId, AZ::ComponentTypeList{ typeId });
+    }
+```
+
+
+## Icon
```

Comment:
This whole section about icons would probably be good to explain, but they don't actually need to add this code. When you create the `ShapeExample` gem from our custom tool gemplate, this code for registering your icon and initializing the resource is already done for you from the template.